ui.apps/src/main/content/jcr_root/apps/venia/clientlibs/common/js/CommerceGraphqlApi.js

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ *
+ *     Copyright 2019 Adobe. All rights reserved.
+ *     This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License. You may obtain a copy
+ *     of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software distributed under
+ *     the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *     OF ANY KIND, either express or implied. See the License for the specific language
+ *     governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+
+(function () {
+
+    'use strict';
+    //todo update the proxy rules so that this gets router through the dispatcher
+    const imageUrlPrefix = "/magento/img";
+
+    class CommerceGraphqlApi {
+
+        constructor(props) {
+            if (!props.endpoint) {
+                throw new Error('The commerce API is not properly initialized. The "endpoint" property is missing from the initialization object');
+            }
+
+            this.endpoint = props.endpoint;
+        }
+
+        async _fetch(url, params) {
+            let response = await fetch(url, params);
+            if (!response.ok || response.errors) {
+                let message = await response.text();
```

Comment:
Hmm, I think you rather separate the 2 error cases, when not ok, this is fine, but when it's a GraphQL error, then it should be extracted from the errors message, see for example https://github.com/adobe/commerce-cif-connector/blob/master/bundles/cif-connector-graphql/src/test/resources/magento-graphql-error.json